### ISSUE TITLE
Aqara Door Contact Sensor naming in Homeassistant adds duplicate "Door Door"

### ISSUE THREAD

**Issue by peteh** [Koenkk/zigbee2mqtt#21092]
### What happened?

Somehow the naming of Aqara Door Contact sensors in Homeassistant  is a bit messed up. 

![image](https://github.com/Koenkk/zigbee2mqtt/assets/918728/9957876a-1be5-456d-a87f-3ab1a1617c7d)

With the new naming scheme changes in Homeassistant Entities are now named
[device name] [sensor]. 

I have a contact sensor on my Washing machine. Therefore I named the sensor in z2m to "Washing Machine Door". Unfortunately the sensor entity that shows if it's closed or not is named "Door" instead of being the main entity of the device and taking its name. 

Thus Homeassistant names the sensor entity to "Washing Machine Door Door". 

With the new naming scheme the name of the contact sensor entity should be set to None/null so that Homeassistant will name it to [device name] removing the duplicate "Door". 

Not sure how to better express that, I hope it's clear what the issue is. 

Z2M Version: [1.35.1](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.35.1) commit: [9eaaa0f](https://github.com/Koenkk/zigbee2mqtt/commit/9eaaa0f)

### What did you expect to happen?

When I name the device to "Washing Machine Door", the entity that shows open/closed should also be named to "Washing Machine Door" and not to "Washing Machine Door Door"

### How to reproduce it (minimal and precise)

Pair an Aquara door sensor and name it to "Something Door". 

### Zigbee2MQTT version

1.35.1

### Adapter firmware version

20220219

### Adapter

zStack3x0

### Setup

Z2M and HA in docker on a mini pc

**Comment by peteh** [comment]
If this is not the right place, can someone point me to where I should report this to please?

**Comment by PaulHam211** [comment]
Issue for me too. I have contact sensors on the windows, I name it 'Bedroom Window', in Home Assistant it gets named "Bedroom Window Door"

**Comment by csszokolai** [comment]
Same issue here.

**Comment by jimb0b** [comment]
Same!

**Comment by greendog99** [comment]
I agree the way this is designed makes naming things difficult / impossible. Only partial solution I've come up with is to manually rename every door (or window) entity to "Contact" so it shows up as "Bedroom Door Contact" instead of "Bedroom Door Door".

**Comment by peteh** [comment]
I doubt it's in HA, it's more likely in z2m. 

The main entity should report null as name but it probably responds with "Door". If it would respond with null, it would get the name of the device. If it responds with "Door" it gets the name of the device + Door. 

So it should be fixable in z2m when it names the entities. However, this code is a bit messy for this device type and I haven't found the obvious fix yet.

**Comment by peteh** [comment]
I tried again to debug it but I could not find out where the additional "Door" comes from. 
In https://github.com/Koenkk/zigbee-herdsman-converters lumi.sensor_magnet.aq2 looks fine to me.

**Comment by scorpio1441** [comment]
Is there a solution to this?

![Image](https://github.com/user-attachments/assets/9ca0503b-629d-4584-bb64-d7be38bdf83d)

**Comment by peteh** [comment]
not stale! There is still no fix

**Comment by peteh** [comment]
Unstale because still broken

**Comment by bucari** [comment]
Same issue for me.

Is it possible to remove the contact suffix in entity name?

I'm migrating from deCONZ to z2m
and have a lot of entities used in automation that i would not edit.

**Comment by peteh** [comment]
Still broken

**Comment by peteh** [comment]
It's still broken

**Comment by peteh** [comment]
Again, still broken

**Comment by peteh** [comment]
Still broken

**Comment by peteh** [comment]
Unstale

**Comment by clemencov** [comment]
Still an issue
